### ISSUE TITLE
experiment_runner.py fails with CUDA accelerator.

### ISSUE THREAD

**Issue by ysiraichi** [pytorch/xla#5943]
## 🐛 Bug

Tried running torchbench with the recently upstreamed script, but couldn't do it using CUDA. Note that I did check that `nvidia-smi` runs without problems.

```bash
python benchmarks/experiment_runner.py --suite-name torchbench --accelerator cuda --test train --dynamo openxla --xla PJRT -k BERT_pytorch
```

Here's what I get, enabling logging.

```
2023-11-29 21:31:40.967649: I torch_xla/csrc/runtime/pjrt_computation_client.cc:133] Initializing PjRt GPU client...
2023-11-29 21:31:40.969522: I external/xla/xla/pjrt/distributed/service.cc:119] Experimental coordination service is enabled.
2023-11-29 21:31:40.970863: I external/xla/xla/pjrt/distributed/service.cc:149] Jax service listening on localhost:8547
2023-11-29 21:31:40.972262: I external/tsl/tsl/distributed_runtime/coordination/coordination_service.cc:553] /job:jax_worker/replica:0/task:0 has connected to coordination service. Incarnation: 1236721036966731102
2023-11-29 21:31:40.972513: I external/tsl/tsl/distributed_runtime/coordination/coordination_service_agent.cc:304] Coordination agent has successfully connected.
2023-11-29 21:31:40.972993: I external/xla/xla/pjrt/distributed/client.cc:132] Connected to distributed JAX controller
2023-11-29 21:31:40.973036: I torch_xla/csrc/runtime/pjrt_computation_client.cc:165] Getting StreamExecutorGpuClient for node_id=0, num_nodes=1
2023-11-29 21:31:41.337790: E external/xla/xla/stream_executor/cuda/cuda_driver.cc:274] failed call to cuInit: CUDA_ERROR_NOT_INITIALIZED: initialization error
2023-11-29 21:31:41.337902: I external/xla/xla/stream_executor/cuda/cuda_diagnostics.cc:129] retrieving CUDA diagnostic information for host: benchmarking-16
2023-11-29 21:31:41.337922: I external/xla/xla/stream_executor/cuda/cuda_diagnostics.cc:136] hostname: benchmarking-16
2023-11-29 21:31:41.338054: I external/xla/xla/stream_executor/cuda/cuda_diagnostics.cc:159] libcuda reported version is: 525.105.17
2023-11-29 21:31:41.338132: I external/xla/xla/stream_executor/cuda/cuda_diagnostics.cc:163] kernel reported version is: NOT_FOUND: could not find kernel module information in driver version file contents: "NVRM version: NVIDIA UNIX Open Kernel Module for x86_64  525.105.17  Release Build  (dvs-builder@U16-T02-30-1)  Tue Mar 28 22:29:15 UTC 2023
GCC version:  gcc version 10.2.1 20210110 (Debian 10.2.1-6)
"
2023-11-29 21:31:41.338601: I external/xla/xla/pjrt/distributed/client.cc:140] Distributed task shutdown initiated.
2023-11-29 21:31:41.338628: I external/tsl/tsl/distributed_runtime/coordination/coordination_service_agent.cc:473] Coordination agent has initiated Shutdown().
2023-11-29 21:31:41.339087: I external/tsl/tsl/distributed_runtime/coordination/coordination_service.cc:1167] Shutdown barrier in coordination service has passed.
2023-11-29 21:31:41.339201: I external/tsl/tsl/distributed_runtime/coordination/coordination_service.cc:658] /job:jax_worker/replica:0/task:0 has disconnected from coordination service.
2023-11-29 21:31:41.339376: I external/tsl/tsl/distributed_runtime/coordination/coordination_service_agent.cc:492] Coordination agent has successfully shut down.
2023-11-29 21:31:41.339540: I external/xla/xla/pjrt/distributed/client.cc:142] Distributed task shutdown result: OK
2023-11-29 21:31:41.339828: I external/xla/xla/pjrt/distributed/service.cc:165] Jax service shutting down
Traceback (most recent call last):
  File "xla/benchmarks/util.py", line 62, in _check_xla_device
    q.put(bool(xm.get_xla_supported_devices(devkind=devkind)))
  File "xla/torch_xla/core/xla_model.py", line 99, in get_xla_supported_devices
    xla_devices = _DEVICES.value
  File "xla/torch_xla/utils/utils.py", line 29, in value
    self._value = self._gen_fn()
  File "xla/torch_xla/core/xla_model.py", line 20, in <lambda>
    _DEVICES = xu.LazyProperty(lambda: torch_xla._XLAC._xla_get_devices())
RuntimeError: Bad StatusOr access: FAILED_PRECONDITION: No visible GPU devices.
WARNING:__main__:Number of selected experiment configs: 0
WARNING:__main__:Number of selected model configs: 1
```

## Environment

- PyTorch/XLA: 402166bec712eaea1d2db9095db4f7e048e70f4a

**Comment by ysiraichi** [comment]
cc @JackCaoG @miladm

**Comment by ysiraichi** [comment]
That's what I thought. So, I tried:
- `CUDA_VISIBLE_DEVICES=0`
- `GPU_NUM_DEVICES=1`

But, none of them worked. I am able to run a simple computation:

```python
a = torch.rand(5, 5, device=xm.xla_device())
r = a @ a
xm.mark_step()
```

Any ideas?

**Comment by JackCaoG** [comment]
no idea.. @vanbasten23 any thoughts?

**Comment by ysiraichi** [comment]
Does it work for you?

**Comment by vanbasten23** [comment]
> no idea.. @vanbasten23 any thoughts?

replied in the PR that Yukio opens

**Comment by ysiraichi** [comment]
Closed: #6068
